File: battleship/cpp/Cruiser.cpp
#include "Cruiser.hpp"
#include <random>
#include <vector>
#include <string>


const int Cruiser::CRUISER_SHOTS = 4;

const int Cruiser::CRUISER_SIZE = 4;

const std::vector<std::string> Cruiser::CRUISER_NAMES({"USS Juneau", "USS Lake Erie", "USS Lake Champlain", "USS Astoria", "USS Indianapolis"});

void Cruiser::init(){
  alive = true;
  shots = CRUISER_SHOTS;
  size = CRUISER_SIZE;
  name = CRUISER_NAMES[rand() % CRUISER_NAMES.size()];
}

File: battleship/java/Cruiser.java
class Cruiser extends Ship{
    
    String[] cruiserNames = {"USS Astoria",
                             "USS Portland",
                             "USS Indianapolis",
                             "USS Juneau",
                             "USS Lake Erie",
                             "USS Augusta"};
    int cruiserSize = 4;
    int cruiserShots = 4;
    
    public Cruiser(){
	init(cruiserNames[(int) (Math.random() * cruiserNames.length)], cruiserSize, cruiserShots);
    }
}
